File: Python/JDI/web/selenium/elements/base/select_element.py
from JDI.web.selenium.elements.base.clickable_text import ClickableText


class SelectElement(ClickableText):
    
    def __init__(self, by_locator=None, web_element=None):
        if by_locator is not None:
            super(SelectElement, self).__init__(by_locator=by_locator)
        elif web_element is not None:
            super(SelectElement, self).__init__(web_element=web_element)
        else:
            super(SelectElement, self).__init__()

File: Java/JDI/jdi-uitest-win/src/main/java/com/epam/jdi/uitests/win/winnium/elements/base/SelectElement.java
package com.epam.jdi.uitests.win.winnium.elements.base;


import com.epam.jdi.uitests.core.interfaces.base.ISelect;
import com.epam.jdi.uitests.win.winnium.elements.base.managers.AttributeTextManager;
import com.epam.jdi.uitests.win.winnium.elements.base.managers.ClickableManager;
import org.openqa.selenium.WebElement;

public class SelectElement extends Element implements ISelect {
    private AttributeTextManager textManager = new AttributeTextManager(this);
    private ClickableManager clickableManager = new ClickableManager(this);

    public SelectElement() {
    }

    public SelectElement(WebElement webElement) {
        super(webElement);
    }

    @Override
    public void select() {
        if (!isSelected())
            click();
    }

    @Override
    public boolean isSelected() {
        return invoker.doJActionResult("Is Selected", () -> getWebElement().isSelected(), this.toString());
    }

    @Override
    public void click() {
        clickableManager.click();
    }

    @Override
    public String getText() {
        return textManager.getText();
    }

    @Override
    public String waitText(String text) {
        return textManager.waitText(text);
    }

    @Override
    public String waitMatchText(String regEx) {
        return textManager.waitMatchText(regEx);
    }

    @Override
    public String getValue() {
        return getText();
    }
}
